src/time.rs

```diff
@@ -264,4 +614,641 @@ mod tests {
+                u: UtcExpectation::new(2019, 12, 31, 23, 59, 60.5),
+                p: Some(make_p_neg_trend()),
+            },
+            TestCase {
+                t: GpsTime::new_unchecked(2086, 259219.0),
+                u: UtcExpectation::new(2020, 01, 01, 00, 00, 00.0),
+                p: Some(make_p_neg_trend()),
+            },
+        ];
+
+        for test_case in test_cases {
+            let expected = &test_case.u;
+            let u = if let Some(p) = &test_case.p {
+                test_case.t.to_utc(p)
+            } else {
+                test_case.t.to_utc_hardcoded()
+            };
+
+            assert_eq!(u.year(), expected.year);
+            assert_eq!(u.month(), expected.month);
+            assert_eq!(u.day_of_month(), expected.day);
+            assert_eq!(u.hour(), expected.hour);
+            assert_eq!(u.minute(), expected.minute);
+            assert!(
+                (u.seconds() - expected.second).abs() < 1e-5,
+                "{} {} {}",
+                u.seconds(),
+                expected.second,
+                test_case.t.tow()
+            );
+        }
+    }
+
+    #[test]
+    fn round_to_epoch() {
+        let soln_freq = 10.0;
+        let epsilon = std::time::Duration::from_secs_f64(1e-5);
+
+        let test_cases = [
+            GpsTime::new_unchecked(1234, 567890.01),
+            GpsTime::new_unchecked(1234, 567890.0501),
+            GpsTime::new_unchecked(1234, 604800.06),
+        ];
+
+        let expectations = [
+            GpsTime::new_unchecked(1234, 567890.00),
+            GpsTime::new_unchecked(1234, 567890.10),
+            GpsTime::new_unchecked(1235, 0.1),
+        ];
+
+        for (test_case, expectation) in test_cases.iter().zip(expectations.iter()) {
+            let rounded = test_case.round_to_epoch(soln_freq);
+            assert!((rounded - expectation) < epsilon);
+        }
+    }
+
+    #[test]
+    fn floor_to_epoch() {
+        let soln_freq = 10.0;
+        let epsilon = std::time::Duration::from_secs_f64(1e-6);
+
+        let test_cases = [
+            GpsTime::new_unchecked(1234, 567890.01),
+            GpsTime::new_unchecked(1234, 567890.0501),
+            GpsTime::new_unchecked(1234, 604800.06),
+        ];
+
+        let expectations = [
+            GpsTime::new_unchecked(1234, 567890.00),
+            GpsTime::new_unchecked(1234, 567890.00),
+            GpsTime::new_unchecked(1235, 0.0),
+        ];
+
+        for (test_case, expectation) in test_cases.iter().zip(expectations.iter()) {
+            let rounded = test_case.floor_to_epoch(soln_freq);
+            assert!((rounded - expectation) < epsilon);
+        }
+    }
+
+    #[test]
+    fn chrono_conversions() {
+        use chrono::prelude::*;
+        let swift_date = UtcTime::from_date(2021, 8, 1, 00, 11, 0.0);
+        let expected_utc = DateTime::<Utc>::from_utc(NaiveDateTime::new(NaiveDate::from_ymd(2021, 8, 1), NaiveTime::from_hms_nano(00, 11, 0, 0)), Utc);
```

Comment:
The CI should fail for unused variables.

Reply:
This is addressed in https://github.com/swift-nav/swiftnav-rs/pull/64